src/renderer/components/item-object-list/item-list-layout.tsx

```diff
@@ -392,19 +392,23 @@ export class ItemListLayout extends React.Component<ItemListLayoutProps> {
   }
 
   renderTableHeader() {
-    const { renderTableHeader, isSelectable, isConfigurable, store } = this.props;
+    const { customizeTableRowProps, renderTableHeader, isSelectable, isConfigurable, store } = this.props;
 
     if (!renderTableHeader) {
       return;
     }
 
+    const enabledItems = customizeTableRowProps
+      ? this.items.filter(item => !customizeTableRowProps(item).disabled)
+      : this.items;
+
     return (
       <TableHead showTopLine nowrap>
         {isSelectable && (
           <TableCell
             checkbox
-            isChecked={store.isSelectedAll(this.items)}
-            onClick={prevDefault(() => store.toggleSelectionAll(this.items))}
+            isChecked={store.isSelectedAll(enabledItems)}
+            onClick={prevDefault(() => store.toggleSelectionAll(enabledItems))}
```

Comment:
I think there's a flaw in the store.toggleSelectionAll()` function, it verifies that each item is selected before deselecting them all. But otherwise it selects them all. This is fine if it is always called with a list of items that are either all selected or all not selected. If not then it should verify that they are all not selected before selecting them all.

Reply:
That does sound like a flaw, but tbh the name doesn't really match either of those behaviours.